src/utils.ts

```diff
@@ -281,3 +292,30 @@ export function smockableNode(node: ASTNode): boolean {
 
   return true;
 }
+
+/**
+ * Renders the abstract functions that are not implemented in the current contract
+ * @param contract The contract to render the abstract functions from
+ * @returns The content of the functions
+ */
+export async function renderAbstractUnimplementedFunctions(contract: ContractDefinition): Promise<string> {
+  let content = '';
+
+  if (contract.vLinearizedBaseContracts.length > 1) {
+    const functions = contract.vFunctions;
+
+    // Skip the first contract, which is the current contract
+    for (let i = 1; i < contract.vLinearizedBaseContracts.length; i++) {
+      const base = contract.vLinearizedBaseContracts[i];
+
+      for (const baseFunction of base.vFunctions) {
+        // If the function is not implemented in the current contract, render it
+        if (!functions.some((func) => func.raw?.functionSelector === baseFunction.raw?.functionSelector)) {
+          content += await renderNodeMock(baseFunction);
+        }
+      }
+    }
+  }
```

Comment:
I have a question, would this be possible?
```typescript
for (const base of contract.vLinearizedBaseContracts) {
    // Skip the first contract, which is the current contract
    if (base === contract) continue;
    //  The rest of the function
```
We could compare by name, canonical name, id or whatever. There are 2 reasons for this, I'm a bit paranoid the `index = 1` thing will be lost during a refactoring or a bug fix and also without it we can remove the `length` check.